frontend/src/metabase/query_builder/actions/core/updateQuestion.ts

```diff
@@ -29,15 +29,23 @@ import { onCloseQuestionInfo, setQueryBuilderMode } from "../ui";
 
 import { getQuestionWithDefaultVisualizationSettings } from "./utils";
 
-function hasNewColumns(question: Question, queryResult: Dataset) {
-  // NOTE: this assume column names will change
+function hasNewColumns(question: Question, queryResult: Dataset | null) {
+  if (!queryResult) {
+    return false;
+  }
+  if (question.isNative()) {
```

Comment:
this change is not clear to me

Reply:
It's due to `query instanceof StructuredQuery` part.